### PR TITLE
CMake build script

### PR DESCRIPTION
**Pull request by ssrlive**
Add CMake build supporting in Android for tun2socks component.

Now we can create a CMakeLists.txt file to use it.
```
cmake_minimum_required(VERSION 3.4)
project("myapplication")
add_subdirectory(libancillary)
add_subdirectory(badvpn)
```
Once the script running finished, we get the file `libtun2socks.so`.

### CODE REVIEW

**Comment by Mygod** [comment]
Thanks for the contribution. However, I recommend submitting this patch to upstream (@ambrop72's repo).

**Comment by ssrlive** [comment]
Since the Android version depends on `libancillary`, I don't think @ambrop72 will accept this patch.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+    set(BUILD_TUN2SOCKS 1 CACHE INTERNAL "BUILD_TUN2SOCKS" )
+    set(HAVE_SYSLOG_H 1 CACHE INTERNAL "HAVE_SYSLOG_H" )
+
+    set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -Wall -Wextra -std=gnu99")
+    include_directories(${LIB_ANCILLARY_DIR})
+
+    add_definitions(-DBADVPN_THREADWORK_USE_PTHREAD -DBADVPN_LINUX -DBADVPN_BREACTOR_BADVPN -D_GNU_SOURCE)
+    add_definitions(-DBADVPN_USE_SIGNALFD -DBADVPN_USE_EPOLL)
+    add_definitions(-DBADVPN_LITTLE_ENDIAN -DBADVPN_THREAD_SAFE)
+    add_definitions(-DNDEBUG -DANDROID -D__ANDROID__)
```
Comment (Mygod) [inline CMakeLists.txt]: By the way, `ANDROID` and `__ANDROID__` should be defined by NDK toolchain.

Comment (ssrlive) [reply CMakeLists.txt]: removed.

**Comment by Mygod** [comment]
> Since the Android version depends on `libancillary`, I don't think @ambrop72 will accept this patch.

Good point. Maybe first submit a PR to base without `libancillary`?

**Comment by ssrlive** [comment]
I have make a [PR](https://github.com/ambrop72/badvpn/pull/123) in origin repo for this patch.

**Pull request closed without merging**